Ryujinx.HLE/HOS/Services/Caps/CaptureManager.cs

```
@@ -0,0 +1,128 @@
+
+                    applicationAlbumEntry = new ApplicationAlbumEntry()
+                    {
+                        Size              = (ulong)Marshal.SizeOf(typeof(ApplicationAlbumEntry)),
+                        TitleId           = titleId,
+                        AlbumFileDateTime = new AlbumFileDateTime()
+                        {
+                            Year     = (ushort)currentDateTime.Year,
+                            Month    = (byte)currentDateTime.Month,
+                            Day      = (byte)currentDateTime.Day,
+                            Hour     = (byte)currentDateTime.Hour,
+                            Minute   = (byte)currentDateTime.Minute,
+                            Second   = (byte)currentDateTime.Second,
+                            UniqueId = 0 // Incremented when there is multiple Album files with the same timestamp. Doesn't occur in our case.
+                        },
+                        AlbumStorage      = AlbumStorage.Sd,
+                        ContentType       = ContentType.Screenshot,
+                        Padding           = new Array5<byte>(),
+                        Unknown0x1f       = 1
+                    };
+
+                    // NOTE: The hex hash is a HMAC-SHA256 (first 32 bytes) using a hardcoded secret key over the titleId, we can simulate it by hashing the titleId instead.
+                    string hash                 = BitConverter.ToString(sha256Hash.ComputeHash(BitConverter.GetBytes(titleId))).Replace("-", "").Remove(0x20);
+                    string fileName             = $"{currentDateTime.ToString("yyyyMMddHHmmss")}{applicationAlbumEntry.AlbumFileDateTime.UniqueId.ToString("00")}-{hash}.jpg";
+                    string screenshotFolderPath = Path.Combine(_sdCardPath, "Nintendo", "Album", currentDateTime.Year.ToString("00"), currentDateTime.Month.ToString("00"), currentDateTime.Day.ToString("00"));
+
+                    Directory.CreateDirectory(screenshotFolderPath);
```

Shouldn't this handle cases where creating the folder fails (due to being out of disk space or whatever)? How does the service handle this?

---

I haven't really REd this part. But it should throw an exception if it's the case right ? Maybe I should do nothing if it reach an exception and log an error ?